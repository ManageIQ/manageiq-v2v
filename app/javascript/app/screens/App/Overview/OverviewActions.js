import { SHOW_MAPPING_WIZARD, SHOW_PLAN_WIZARD } from './OverviewConstants';

export const showMappingWizardAction = () => dispatch => {
  dispatch({
    type: SHOW_MAPPING_WIZARD
  });
};

export const showPlanWizardAction = () => dispatch => {
  dispatch({
    type: SHOW_PLAN_WIZARD
  });
};
