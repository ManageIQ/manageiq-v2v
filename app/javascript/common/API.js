import axios from 'axios';

const getAuthToken = () =>
  localStorage.miq_token ? localStorage.miq_token : '';

axios.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest';
axios.defaults.headers.common['X-Auth-Token'] = getAuthToken();

export default {
  get(url, headers = {}, params = {}) {
    return axios.get(url, {
      headers,
      params
    });
  },
  put(url, data = {}, headers = {}) {
    return axios.put(url, data, {
      headers
    });
  },
  post(url, data = {}, headers = {}) {
    return axios.post(url, data, {
      headers
    });
  },
  delete(url, headers = {}) {
    return axios.delete(url, {
      headers
    });
  },
  patch(url, data = {}, headers = {}) {
    return axios.patch(url, data, {
      headers
    });
  }
};

export const globalMockMode = true;
