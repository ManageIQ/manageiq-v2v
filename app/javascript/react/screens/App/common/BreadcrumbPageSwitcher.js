import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { OverlayTrigger, Popover, Button, Icon, ListGroup, ListGroupItem } from 'patternfly-react';

const BreadcrumbPageSwitcher = ({ activeHref }) => {
  const SwitcherItem = ({ name, href }) => (
    <ListGroupItem className={classNames('no-border', { active: href === activeHref })} href={href}>
      {name}
    </ListGroupItem>
  );
  SwitcherItem.propTypes = {
    name: PropTypes.string,
    href: PropTypes.string
  };

  // TODO: remove this when we are ready to release analytics
  const hideMigrationAnalytics = true;

  const popoverContent = (
    <ListGroup style={{ marginBottom: 0, borderTop: 0 }}>
      <SwitcherItem name={__('Migration Plans')} href="#/plans" />
      <SwitcherItem name={__('Infrastructure Mappings')} href="#/mappings" />
      <SwitcherItem name={__('Migration Settings')} href="#/settings" />
      {!hideMigrationAnalytics && <SwitcherItem name={__('Migration Analytics')} href="#/analytics" />}
    </ListGroup>
  );

  return (
    <OverlayTrigger
      rootClose
      trigger="click"
      placement="bottom"
      overlay={<Popover id="breadcrumb-switcher-popover">{popoverContent}</Popover>}
    >
      <Button
        style={{
          marginTop: '-2px',
          marginBottom: '-4px',
          marginLeft: '15px',
          height: '22px',
          width: '32px',
          lineHeight: '14px'
        }}
      >
        <Icon type="fa" name="exchange" />
      </Button>
    </OverlayTrigger>
  );
};

BreadcrumbPageSwitcher.propTypes = {
  activeHref: PropTypes.string
};

export default BreadcrumbPageSwitcher;
