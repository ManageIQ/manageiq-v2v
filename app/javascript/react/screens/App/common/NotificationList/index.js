import { connect } from 'react-redux';
import NotificationList from './NotificationList';
import * as NotificationListActions from './NotificationListActions';
import reducer from './NotificationListReducer';

export const reducers = { notifications: reducer };

const mapStateToProps = ({ notifications }, ownProps) => ({
  ...notifications,
  ...ownProps.data
});

const mergeProps = (stateProps, dispatchProps, ownProps) =>
  Object.assign(stateProps, ownProps.data, dispatchProps);

export default connect(mapStateToProps, NotificationListActions, mergeProps)(
  NotificationList
);
