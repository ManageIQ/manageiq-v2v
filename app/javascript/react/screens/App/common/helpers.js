import React from 'react';

// TODO remove these, they are space fillers
const t = str => (
  <div align="center">
    <h1>TODO: {str}!</h1>
  </div>
);

export const todo = str => (
  <div>
    {t(str)}
    {t(str)}
    {t(str)}
    {t(str)}
    {t(str)}
  </div>
);
