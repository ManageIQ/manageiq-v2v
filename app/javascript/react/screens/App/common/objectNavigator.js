export default (path, object) =>
  path.split('.').reduce((previous, current) => (previous ? previous[current] : null), object);
