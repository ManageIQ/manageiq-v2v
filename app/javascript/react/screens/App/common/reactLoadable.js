import React from 'react';
import PropTypes from 'prop-types';
import Loadable from 'react-loadable';

const LoadingPage = ({ error, pastDelay, timedOut }) => {
  if (error) {
    return <div>Error! Please refresh and try again.</div>;
  } else if (timedOut) {
    return <div>Timed out. Please refresh and try again.</div>;
  } else if (pastDelay) {
    return <div>Loading...</div>;
  }
  return null;
};

export default options =>
  Loadable({
    ...options,
    loading: LoadingPage,
    delay: 200,
    timeout: 10000
  });

LoadingPage.propTypes = {
  error: PropTypes.bool,
  pastDelay: PropTypes.bool,
  timedOut: PropTypes.bool
};
