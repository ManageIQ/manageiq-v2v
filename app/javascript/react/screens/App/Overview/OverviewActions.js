import URI from 'urijs';
import API, { globalMockMode } from '../../../../common/API';

import {
  SHOW_MAPPING_WIZARD,
  SHOW_PLAN_WIZARD,
  HIDE_MAPPING_WIZARD,
  FETCH_V2V_TRANSFORMATION_MAPPINGS,
  FETCH_V2V_TRANSFORMATION_PLAN_REQUESTS,
  FETCH_V2V_TRANSFORMATION_PLANS
} from './OverviewConstants';

import { requestTransformationMappingsData } from './overview.fixtures';
import { requestTransformationPlanRequestsData } from './overview.transformationPlanRequests.fixtures';
import { requestTransformationPlansData } from './overview.transformationPlans.fixtures';

const mockMode = globalMockMode;

export const showMappingWizardAction = () => dispatch => {
  dispatch({
    type: SHOW_MAPPING_WIZARD
  });
};

export const showPlanWizardAction = id => dispatch => {
  dispatch({
    type: SHOW_PLAN_WIZARD,
    payload: id
  });
};

const _getTransformationMappingsActionCreator = url => dispatch => {
  dispatch({
    type: FETCH_V2V_TRANSFORMATION_MAPPINGS,
    payload: API.get(url)
  }).catch(error => {
    // to enable UI development without the backend ready, i'm catching the error
    // and passing some mock data thru the FULFILLED action after the REJECTED action is finished.
    if (mockMode) {
      dispatch({
        type: `${FETCH_V2V_TRANSFORMATION_MAPPINGS}_FULFILLED`,
        payload: requestTransformationMappingsData.response
      });
    }
  });
};

export const fetchTransformationMappingsAction = url => {
  const uri = new URI(url);
  return _getTransformationMappingsActionCreator(uri.toString());
};

const _getTransformationPlanRequestsActionCreator = url => dispatch =>
  dispatch({
    type: FETCH_V2V_TRANSFORMATION_PLAN_REQUESTS,
    payload: API.get(url)
  }).catch(error => {
    // redux-promise-middleware will automatically send:
    // FETCH_V2V_TRANSFORMATION_PLAN_REQUESTS_PENDING, FETCH_V2V_TRANSFORMATION_PLAN_REQUESTS_FULFILLED,
    // FETCH_V2V_TRANSFORMATION_PLAN_REQUESTS_REJECTED

    // to enable UI development without the database, i'm catching the error
    // and passing some mock data thru the FULFILLED action after the REJECTED action is finished.
    if (mockMode) {
      dispatch({
        type: `${FETCH_V2V_TRANSFORMATION_PLAN_REQUESTS}_FULFILLED`,
        payload: requestTransformationPlanRequestsData.response
      });
    }
  });

export const fetchTransformationPlanRequestsAction = url => {
  const uri = new URI(url);
  return _getTransformationPlanRequestsActionCreator(uri.toString());
};

const _getTransformationPlansActionCreator = url => dispatch =>
  dispatch({
    type: 'FETCH_V2V_TRANSFORMATION_PLANS',
    payload: API.get(url)
  }).catch(error => {
    // redux-promise-middleware will automatically send:
    // FETCH_V2V_TRANSFORMATION_PLANS_PENDING, FETCH_V2V_TRANSFORMATION_PLANS_FULFILLED,
    // FETCH_V2V_TRANSFORMATION_PLANS_REJECTED

    // to enable UI development without the database, i'm catching the error
    // and passing some mock data thru the FULFILLED action after the REJECTED action is finished.
    if (mockMode) {
      dispatch({
        type: `${FETCH_V2V_TRANSFORMATION_PLANS}_FULFILLED`,
        payload: requestTransformationPlansData.response
      });
    }
  });

export const fetchTransformationPlansAction = url => {
  const uri = new URI(url);
  return _getTransformationPlansActionCreator(uri.toString());
};

export const continueToPlanAction = id => dispatch => {
  dispatch({
    type: HIDE_MAPPING_WIZARD
  });
  dispatch({
    type: SHOW_PLAN_WIZARD,
    payload: { id }
  });
};
