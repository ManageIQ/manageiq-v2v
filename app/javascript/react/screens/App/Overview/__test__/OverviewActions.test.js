import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { mockRequest, mockReset } from '../../../../../common/mockRequests';

import { fetchTransformationPlanRequestsAction } from '../OverviewActions';
import { requestTransformationPlanRequestsData } from '../overview.transformationPlanRequests.fixtures';

const middlewares = [thunk, promiseMiddleware()];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

afterEach(() => {
  store.clearActions();
  mockReset();
});

describe('fetchTransformationPlanRequestsAction', () => {
  const {
    fetchTransformationPlanRequestsUrl
  } = requestTransformationPlanRequestsData;
  const request = { fetchTransformationPlanRequestsUrl };

  test('dispatches PENDING and FULFILLED actions', () => {
    mockRequest({
      ...request,
      status: 200
    });

    return store
      .dispatch(
        fetchTransformationPlanRequestsAction(
          fetchTransformationPlanRequestsUrl
        )
      )
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });

  test('dispatches PENDING and REJECTED actions', () => {
    mockRequest({
      ...request,
      status: 404
    });

    return store
      .dispatch(
        fetchTransformationPlanRequestsAction(
          fetchTransformationPlanRequestsUrl
        )
      )
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});
