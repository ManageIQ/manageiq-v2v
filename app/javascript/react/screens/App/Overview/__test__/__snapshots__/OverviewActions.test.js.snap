// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchTransformationPlanRequestsAction dispatches PENDING and FULFILLED actions 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_TRANSFORMATION_PLAN_REQUESTS_PENDING",
  },
  Object {
    "payload": Object {
      "config": Object {
        "adapter": null,
        "data": undefined,
        "headers": Object {
          "Accept": "application/json, text/plain, */*",
          "X-Auth-Token": "",
          "X-Requested-With": "XMLHttpRequest",
        },
        "maxContentLength": -1,
        "method": "get",
        "params": Object {},
        "timeout": 0,
        "transformRequest": Object {
          "0": [Function],
        },
        "transformResponse": Object {
          "0": [Function],
        },
        "url": "/api/dummyPlanRequests",
        "validateStatus": [Function],
        "xsrfCookieName": "XSRF-TOKEN",
        "xsrfHeaderName": "X-XSRF-TOKEN",
      },
      "data": null,
      "headers": undefined,
      "status": 200,
    },
    "type": "FETCH_V2V_TRANSFORMATION_PLAN_REQUESTS_FULFILLED",
  },
]
`;

exports[`fetchTransformationPlanRequestsAction dispatches PENDING and REJECTED actions 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_TRANSFORMATION_PLAN_REQUESTS_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: Request failed with status code 404],
    "type": "FETCH_V2V_TRANSFORMATION_PLAN_REQUESTS_REJECTED",
  },
]
`;

exports[`fetchTransformationPlansAction dispatches PENDING and FULFILLED actions 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_TRANSFORMATION_PLANS_PENDING",
  },
  Object {
    "payload": Object {
      "config": Object {
        "adapter": null,
        "data": undefined,
        "headers": Object {
          "Accept": "application/json, text/plain, */*",
          "X-Auth-Token": "",
          "X-Requested-With": "XMLHttpRequest",
        },
        "maxContentLength": -1,
        "method": "get",
        "params": Object {},
        "timeout": 0,
        "transformRequest": Object {
          "0": [Function],
        },
        "transformResponse": Object {
          "0": [Function],
        },
        "url": "/api/dummyTransformationPlans",
        "validateStatus": [Function],
        "xsrfCookieName": "XSRF-TOKEN",
        "xsrfHeaderName": "X-XSRF-TOKEN",
      },
      "data": null,
      "headers": undefined,
      "status": 200,
    },
    "type": "FETCH_V2V_TRANSFORMATION_PLANS_FULFILLED",
  },
]
`;

exports[`fetchTransformationPlansAction dispatches PENDING and REJECTED actions 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_TRANSFORMATION_PLANS_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: Request failed with status code 404],
    "type": "FETCH_V2V_TRANSFORMATION_PLANS_REJECTED",
  },
]
`;
